sdk/go/pulumi/types.go

```diff
@@ -256,9 +257,12 @@ func (o *OutputState) await(ctx context.Context) (interface{}, bool, bool, []Res
 			}
 			o.cond.Wait()
 		}
-		o.cond.L.Unlock()
 
+		// Make sure o.deps is accessed under the o.cond.L.Lock to avoid torn slice panics.
 		deps = mergeDependencies(deps, o.deps)
+
+		o.cond.L.Unlock()
```

Comment:
Oh nice catch

Reply:
We do not have evidence there is a race here or it relates to the customer issue, in fact it seems it does not, but better safe then sorry. The change makes o.deps access happen under lock.